core/src/main/java/org/pentaho/platform/web/http/api/resources/GenericFileResource.java

```diff
@@ -54,24 +54,37 @@ public GenericFileResource( @NonNull IGenericFileService genericFileService ) {
   }
 
   @GET
-  @Path( "/folders/tree" )
+  @Path( "/files/tree" )
   @Produces( { MediaType.APPLICATION_JSON } )
   @StatusCodes( {
     @ResponseCode( code = 200, condition = "Operation successful" ),
     @ResponseCode( code = 401, condition = "Authentication required" ),
     @ResponseCode( code = 403, condition = "Access forbidden" ),
     @ResponseCode( code = 500, condition = "Operation failed" )
   } )
-  public IGenericFileTree getFolderTree( @QueryParam( "depth" ) Integer maxDepth,
-                                         @QueryParam( "expandedPath" ) String expandedPath ) {
+  public IGenericFileTree getFileTree( @QueryParam( "depth" ) Integer maxDepth,
+                                       @QueryParam( "expandedPath" ) String expandedPath,
+                                       @QueryParam( "filter" ) String treeFilterString ) {
```

Comment:
Perhaps, simply `filter`, like with the other parameters, matching the name of the corresponding `GetTreeOptions` property. Then, a new overload of `GetTreeOptions#setFilter(String) throws IllegalArgumentException;` could be added to centralize and handle the standard parsing. This is done similarly for the string path values of the base and expanded path properties.